packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -997,3 +1246,1035 @@ class _RenderTheatre extends RenderBox with ContainerRenderObjectMixin<RenderBox
     ];
   }
 }
+
+
+// * OverlayPortal Implementation
+//  OverlayPortal is inspired by the
+//  [flutter_portal](https://pub.dev/packages/flutter_portal) package.
+//
+// ** RenderObject hierarchy
+// The widget works by inserting its overlay child's render subtree directly
+// under [Overlay]'s render object (_RenderTheatre).
+// https://user-images.githubusercontent.com/31859944/171971838-62ed3975-4b5d-4733-a9c9-f79e263b8fcc.jpg
+//
+// To ensure the overlay child render subtree does not do layout twice, the
+// subtree must only perform layout after both its _RenderTheatre and the
+// [OverlayPortal]'s render object (_RenderLayoutSurrogateProxyBox) have
+// finished layout. This is handled by _RenderDeferredLayoutBox.
```

Comment:
This sounds to me like it should be possible for the subtree to lay itself out based on the constraints or size of its _RenderTheatre parent.  How would that be done?  Maybe you should include another example that shows how to do that.

Reply:
Talked offline: With this PR, could use a LayoutBuilder to get the constraints with no 1 frame delay.  BUT still need to use a CompositedTransformFollower.